ci_framework/roles/libvirt_manager/tasks/attach_interface.yml

```diff
@@ -14,21 +14,41 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 
-
 # This task requires
 #   vm_name         Domain name to which the interface needs to be attached.
-#   interface_name  source of the device
-#   interface_type  bridge or network
+#   network
+#     name          Name of the network bridge
+#     cidr          network notation
+#     mtu           MTU applied on the interface.
+#     static_ip     boolean if static address is required.
+
+
+- name: Generate a random MAC address
+  ansible.builtin.set_fact:
+    _lm_mac_address: "{{ '0A:02' | community.general.random_mac }}"
+
+- name: Reserve an IP address
+  when: network.static_ip | bool
+  vars:
+    mac_address: "{{ _lm_mac_address }}"
+  ansible.builtin.import_tasks: reserve_ip.yml
 
 - name: Attach interface to the virtual machine.
-  ci_script:
-    output_dir: "{{ cifmw_libvirt_manager_basedir }}/artifacts"
-    script: >-
+  vars:
+    _net_name: >
+      {{
+        'cifmw-net_' + network.name
```

Comment:
networks as managed by libvirt_manager use `cifmw-` prefix, not `cifmw-net_`

Reply:
I had switch to using `cifmw-net_` for those virtual networks that would be created from Hybrid side. This line tends to replace the name of bridge https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci_framework/roles/libvirt_manager/tasks/deploy_layout.yml#L24 

Hence the new prefix. When the network name comes from the reproducer I believe it would automatically have it

In the case of hybrid, there is no DHCP supported in the libvirt network.

Reply:
oh, ok, so that's NOT related to the layout created by the libvirt_manager. OK, I get it.